### PR TITLE
Aruba map

### PR DESCRIPTION
**Pull request by jcfilben**
Adds map for aruba use case.

To do:
- [ ] add more content on the page around the map (search, filter, etc)
- [ ] revisit Pin styling. Currently the pins are similar in size to the smaller clusters. The pin also looks too round.

### CODE REVIEW

File: demo-app/src/components/aruba/ArubaPopup.jsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      <Text color="text-strong" size="xsmall">
+        {location.status[status]} {status}
+      </Text>
+    </Box>
+  );
+};
+
+export const ArubaPopup = ({ location }) => {
+  return (
+    <Popup>
```
Comment (taysea) [inline demo-app/src/components/aruba/ArubaPopup.jsx:19]: Small comment. But I wonder if the caller shouldn't have to import "Popup" and this should just come for free with the `popup` prop. Similar to what we have in MarkerCluster.

Comment (jcfilben) [reply demo-app/src/components/aruba/ArubaPopup.jsx]: Yes, good idea! Just pushed up that change

File: demo-app/src/components/aruba/ArubaPin.jsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import { Box, Text, ThemeContext } from 'grommet';
+import { normalizeTheme } from 'grommet-leaflet-core/src/utils';
```
Comment (taysea) [inline demo-app/src/components/aruba/ArubaPin.jsx:3]: If we are expecting callers to use this util (which seems reasonable given this use case), we might want to think of a better export structure so they don't have to drill in so far

File: demo-app/src/components/aruba/ArubaPin.jsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+const StyledContent = styled(Box)`
+  // multiplier of font-size, for tighter alignment
+  line-height: 1rem;
+  // style needed to offset for rotation of location pin so
+  // icon is vertical
+  transform: rotate(-45deg);
+`;
+
+const ArubaPin = ({ kind = 'default', type }) => {
```
Comment (taysea) [inline demo-app/src/components/aruba/ArubaPin.jsx:25]:
```suggestion
const ArubaPin = ({ kind = 'default', label }) => {
```
Soft opinion

File: demo-app/src/components/aruba/ArubaPin.jsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import { Box, Text, ThemeContext } from 'grommet';
+import { normalizeTheme } from 'grommet-leaflet-core/src/utils';
+import styled from 'styled-components';
+
+const StyledBox = styled(Box)`
```
Comment (taysea) [inline demo-app/src/components/aruba/ArubaPin.jsx:6]: Doesn't have to be solved for here, but might be nice to have an easier way for callers to use this css snippet from grommet-leaflet-core

File: demo-app/src/components/aruba/ArubaData.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+const type = ['Branch', 'Campus', 'Main', 'Data center'];
```
Comment (halocline) [inline demo-app/src/components/aruba/ArubaData.js:1]: I like having all of these dependent files in a single directory -- keeps the domain confined.

A couple suggestions:
- Let's genericize the branding by replacing "aruba" references with "networking"
- create a `data` sub directory even if its a single file
- reserve Title case file names for components

File: demo-app/src/pages/aruba.jsx [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+import { Anchor, Page, PageContent, PageHeader, Footer } from 'grommet';
+import ArubaMap from '../components/aruba/ArubaMap';
+
+const Index = () => {
+  return (
+    <Page kind="full" fill>
+      <PageContent flex>
+        <PageHeader title="Map" />
+        <ArubaMap />
+        <Footer pad="small">
```
Comment (taysea) [inline demo-app/src/pages/aruba.jsx:11]: I think we can remove the Footer.

File: demo-app/package.json [hunk 7]
```diff
@@ -18,7 +18,8 @@
     "grommet-theme-hpe": "^5.0.3",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
-    "react-router-dom": "^6.13.0"
+    "react-router-dom": "^6.13.0",
+    "styled-components": "^5.3.10"
```
Comment (halocline) [inline demo-app/package.json:22]: should this be moved to the root level package.json since it is shared across workspaces? Remove from each local workspace?

File: demo-app/src/components/aruba/ArubaMap.jsx [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+  Map,
+  Marker,
+  MarkerCluster,
+  Pin,
+} from 'grommet-leaflet-core';
+
+import { userLocation, formatLocationsToLatLng } from '../../utils/locations';
+import { generateArubaLocations } from './ArubaData';
+import { ClusterPopup } from '../../ClusterPopup';
+import { ArubaPin } from './ArubaPin';
```
Comment (taysea) [inline demo-app/src/components/aruba/ArubaMap.jsx:16]: Can we add an `index.js` file in this repo so we can import for `'.'`?

Also, to keep things more generic, can we remove all the "Aruba" references and rename to "network"? (Including the pages/aruba.jsx, I forgot to make that one generic)

File: demo-app/src/components/aruba/ArubaMap.jsx [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+          zoomControl={false}
+          theme={hpeLeaflet}
+        >
+          <Controls locations={formatLocationsToLatLng(locations)} />
+          <Marker position={geolocation} icon={<Grommet />} />
+          <MarkerCluster
+            popup={cluster => <ClusterPopup cluster={cluster} />}
+            icon={cluster => {
+              const kind = getClusterStatus(cluster.getAllChildMarkers());
+              const size = getClusterSize(cluster);
```
Comment (taysea) [inline demo-app/src/components/aruba/ArubaMap.jsx:52]: From a visual standpoint, the clusters are the same size as the pins which makes it difficult to scan. I'm wondering if we should use the "medium" cluster in this case so it's bigger than the individual pins?

File: demo-app/src/pages/aruba.jsx [hunk 10]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { Anchor, Page, PageContent, PageHeader, Footer } from 'grommet';
+import ArubaMap from '../components/aruba/ArubaMap';
+
+const Index = () => {
+  return (
+    <Page kind="full" fill>
+      <PageContent flex>
+        <PageHeader title="Map" />
```
Comment (taysea) [inline demo-app/src/pages/aruba.jsx:9]:
```suggestion
        <PageHeader title="Networks" />
```

File: demo-app/src/components/aruba/ArubaMap.jsx [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+
+  return (
+    <Box ref={containerRef} flex background="background-contrast">
+      {geolocation && (
+        <Map
+          id="map"
+          ref={mapContainerRef}
+          center={geolocation}
+          zoom={6}
+          zoomControl={false}
```
Comment (halocline) [inline demo-app/src/components/aruba/ArubaMap.jsx:43]: Let's try to minimize the number props specified here. I believe these three can be removed b/c Map now has a greater number of defaults.

File: demo-app/src/components/aruba/ArubaMap.jsx [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+          theme={hpeLeaflet}
+        >
+          <Controls locations={formatLocationsToLatLng(locations)} />
+          <Marker position={geolocation} icon={<Grommet />} />
```
Comment (halocline) [inline demo-app/src/components/aruba/ArubaMap.jsx:47]: Is this relevant to this example / use case?

```suggestion
```

File: demo-app/src/components/aruba/ArubaMap.jsx [hunk 13]
```diff
@@ -0,0 +1,86 @@
...
+              return <Cluster kind={kind} size={size} />;
+            }}
+          >
+            {locations.map((location, index) => {
+              const total =
+                location.status.critical +
+                location.status.warning +
+                location.status.good;
+              let status = 'good';
+              if (total * 0.15 < location.status.critical) {
```
Comment (halocline) [inline demo-app/src/components/aruba/ArubaMap.jsx:63]: Avoid hard coded expressions?

```suggestion
              const THRESHOLD = 0.15;
              if (total * THRESHOLD < location.status.critical) {
```

File: demo-app/src/pages/aruba.jsx [hunk 14]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
```
Comment (halocline) [inline demo-app/src/pages/aruba.jsx:1]: Related to my directory structure comment earlier, what about the following which will keep all related files contained?

/pages
|__ /network-devices
&nbsp;&nbsp;|__ /data
  &nbsp;&nbsp;&nbsp;&nbsp;|__ network-device-data.js  
&nbsp;&nbsp;|__ index.jsx // actual page
&nbsp;&nbsp;|__ NetworkMap.jsx
&nbsp;&nbsp;|__ NetworkPin.jsx
&nbsp;&nbsp;

**Review by halocline — COMMENTED** [review]
Looking good. Just some comments regarding proposed project structure.

File: demo-app/src/components/aruba/ArubaPopup.jsx [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+        Location:{' '}
+        <Text size="xsmall" weight="normal">
+          San Francisco, CA
+        </Text>
+      </Text>
+      <Text color="text-strong" weight={500} size="xsmall">
+        Type:{' '}
+        <Text size="xsmall" weight="normal">
+          {location.type}
+        </Text>
```
[the remark this answers is not in the document]
Comment (jcfilben) [reply demo-app/src/components/aruba/ArubaPopup.jsx:34]: I was having a hard time getting NameValueList to look right in the space without overriding the hpe theme

File: demo-app/src/components/aruba/ArubaPin.jsx (hunk 2 above)
Comment (jcfilben) [reply demo-app/src/components/aruba/ArubaPin.jsx]: Adjusted export structure in latest commit

File: demo-app/src/components/aruba/ArubaPin.jsx (hunk 3 above)
Comment (jcfilben) [reply demo-app/src/components/aruba/ArubaPin.jsx]: Changed

File: demo-app/src/pages/aruba.jsx (hunk 6 above)
Comment (jcfilben) [reply demo-app/src/pages/aruba.jsx]: Done

File: demo-app/src/components/aruba/ArubaMap.jsx (hunk 8 above)
Comment (jcfilben) [reply demo-app/src/components/aruba/ArubaMap.jsx]: Ended up changing file structure (see Matt's comments)

File: demo-app/src/components/aruba/ArubaMap.jsx (hunk 9 above)
Comment (jcfilben) [reply demo-app/src/components/aruba/ArubaMap.jsx]: Changed to 'medium'

File: demo-app/src/pages/aruba.jsx (hunk 10 above)
Comment (jcfilben) [reply demo-app/src/pages/aruba.jsx]: Done

File: demo-app/src/components/aruba/ArubaData.js (hunk 5 above)
Comment (jcfilben) [reply demo-app/src/components/aruba/ArubaData.js]: Done

File: demo-app/package.json (hunk 7 above)
Comment (jcfilben) [reply demo-app/package.json]: Done

File: demo-app/src/components/aruba/ArubaMap.jsx (hunk 11 above)
Comment (jcfilben) [reply demo-app/src/components/aruba/ArubaMap.jsx]: Done

File: demo-app/src/components/aruba/ArubaMap.jsx (hunk 12 above)
Comment (jcfilben) [reply demo-app/src/components/aruba/ArubaMap.jsx]: No, removed

File: demo-app/src/components/aruba/ArubaMap.jsx (hunk 13 above)
Comment (jcfilben) [reply demo-app/src/components/aruba/ArubaMap.jsx]: Done

File: demo-app/src/pages/aruba.jsx (hunk 14 above)
Comment (jcfilben) [reply demo-app/src/pages/aruba.jsx]: Done

**Review by taysea — COMMENTED** [review]
Tiny comments,

1. Can we add a ReverseAnchor back to home in the page header?
2. Can we wrap the Map in a `ContentContainer` for consistency with other pages?

**Review by taysea — APPROVED** [review]
